arch/arm/core/thread.c

```diff
@@ -67,22 +67,23 @@ void _new_thread(struct k_thread *thread, k_thread_stack_t *stack,
 	pInitCtx = (struct __esf *)(STACK_ROUND_DOWN(stackEnd -
 						     sizeof(struct __esf)));
 
+#if CONFIG_ARM_USERSPACE
+	if (!(options & K_USER)) {
+		pInitCtx->pc = ((u32_t)_thread_entry) & 0xfffffffe;
```

Comment:
why the mask on the function pointer?

Reply:
I figured it out.  It's forcing the mode to ARM mode.  If the LSB is 1, it means it's thumb mode.